framework/validators/UniqueValidator.php

```diff
@@ -194,6 +194,15 @@ private function modelExists($targetClass, $conditions, $model)
                 
                 // any with relation can't be loaded because related fields are not selected
                 $query->with = null;
+    
+                if (is_array($query->joinWith)) {
```

Comment:
Is there a need to take non-eager-loaded `joinWith` or `with` into account when building a query for uniqueness validation at all?

Reply:
My use case requires a external table holding an account id, I join on this to prevent unique collision within an account but allow duplicates within the table structure itself.